package-coverage/generator/generator.go

```diff
@@ -78,10 +78,35 @@ type Generator struct {
 
 	// Tags is arguments passed to the go test runner
 	Tags string
+
+	// SequentialMode controls if the tests are run in parallel or not.  Original version was sequential
+	SequentialMode bool
 }
 
 func (g *Generator) do(paths []string) {
-	for _, path := range paths {
-		generateCoverage(path, g.Exclusion, g.QuietMode, g.Tags)
+	if g.SequentialMode {
```

Comment:
I would create a variable with something like `concurrency`, and set it to 1 for sequential, and `runtime.NumCPUs()` otherwise.

Subsequently, I would spawn `concurrency` number of workers, that reads from an input channel.

Might be overengineering, but might be cleaner. :)

Reply:
Yeah, feels like over but good engineering :)